### PR TITLE
Remove python-small workflow

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Remove the `python-small` workflow because it's a subset of the `python-large` workflow.

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — COMMENTED** [review]
@harupy can we update `CONTRIBUTING.rst` to remove references to small tests?

File: dev/run-large-python-tests.sh [hunk 1]
```diff
@@ -6,7 +6,6 @@ err=0
 trap 'err=1' ERR
```
Comment (dbczumar) [inline dev/run-large-python-tests.sh:6]: Shall we rename this script & the CI job to `run-python-tests`?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -402,17 +402,7 @@ Then, verify that the unit tests & linter pass before submitting a pull request
...
-    ./dev/run-small-python-tests.sh
-    # Optionally, run large tests as well. Github actions will run large tests on your pull request once
-    # small tests pass. Note: models and model deployment tests are considered "large" tests. If
-    # making changes to these components, we recommend running the relevant tests (e.g. tests under
-    # tests/keras for changes to Keras model support) locally before submitting a pull request.
-    ./dev/run-large-python-tests.sh
-
-Python tests are split into "small" & "large" categories, with new tests falling into the "small"
-category by default. Tests that take 10 or more seconds to run should be marked as large tests
-via the ``@pytest.mark.large`` annotation. Dependencies for small and large tests can be added to
```
Comment (harupy) [inline CONTRIBUTING.rst:414]: Can we remove `@pytest.mark.large`?

Comment (dbczumar) [reply CONTRIBUTING.rst:414]: Yep!

File: dev/run-large-python-tests.sh (hunk 1 above)
Comment (harupy) [reply dev/run-large-python-tests.sh]: renamed!

File: CONTRIBUTING.rst (hunk 2 above)
Comment (harupy) [reply CONTRIBUTING.rst:414]: removed all `pytest.mark.large`!

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**